src/ClientBuilder.php

```diff
@@ -124,11 +90,11 @@
     /**
      * Class constructor.
      *
-     * @param array $options The client options
+     * @param Options $options The client options
      */
-    public function __construct(array $options = [])
+    public function __construct(Options $options = null)
```

Comment:
Since this argument is `nullable` you should use the new PHP notation with the question mark before the typehint

Reply:
That's not the same.
 * `Options $options = null` accepts `Options` or nothing, and defaults to `null`
 * `?Options $options` accepts `Options` or `null`, doesn't accept nothing
 * `?Options $options = null` accepts `Options`, `null` or nothing, and defaults to `null`

Reply:
I was thinking more about the tests where we may want to test all the cases (passing options or using the default ones) and using a dataprovider this would not be possible if the argument is not `nullable` :thinking:

Reply:
That's not an issue. I've added such test in 1e2c94d